examples/docker.yaml

```diff
@@ -45,7 +45,20 @@ provision:
     set -eux -o pipefail
     command -v docker >/dev/null 2>&1 && exit 0
     export DEBIAN_FRONTEND=noninteractive
-    curl -fsSL https://get.docker.com | sh
+
+    # DOCKER_INSTALL_START
+    # From official Docker Engine installation guide: https://docs.docker.com/engine/install/ubuntu/#install-using-the-repository
+    apt-get -y install ca-certificates curl gnupg lsb-release
+
+    mkdir -p /etc/apt/keyrings
+    curl -fsSL https://download.docker.com/linux/ubuntu/gpg | gpg --dearmor -o /etc/apt/keyrings/docker.gpg
```

Comment:
Is this really more secure than `curl -fsSL https://get.docker.com | sh` ?

Reply:
(I assume difficulty of pwning download.docker.com is same as pwning get.docker.com)

Reply:
My intention was not about avoiding pwning, the official doc says "It's not recommanded for production environment". There was no clear explanation in the official page, but I assume the script itself is not maintained as production ready by the devs.